### PR TITLE
[GUI] Added config menu button and moved options dialog into it's own page

### PR DESCRIPTION
**Pull request by mxaddict**

### CODE REVIEW

**Comment by navbuilder** [comment]
A new build of 2533d499cffabed45cb039b6a51b41a4db3144fd has completed succesfully!
Binaries available at https://build.nav.community/binaries/moved-options-dialog

**Comment by mxaddict** [comment]
Not sure I have the time to finish this PR as of late, if anyone wants to takeover, feel free to :)

**Comment by mxaddict** [comment]
Re-opening, I think I can get on this now

**Comment by navbuilder** [comment]
A new build of 59bffed05fa15dbeb38410a1014c030edad48d54 has completed succesfully!
Binaries available at https://build.nav.community/binaries/moved-options-dialog

**Comment by navbuilder** [comment]
A new build of 039a1f06559a9979ad4f8aa656e6fdff686c6391 has completed succesfully!
Binaries available at https://build.nav.community/binaries/moved-options-dialog

**Comment by chasingkirkjufell** [comment]
image
image
image
image
image

**Comment by mxaddict** [comment]
@chasingkirkjufell did you find any issues on this branch?

**Comment by chasingkirkjufell** [comment]
I am just thinking if this is what we want? What do you think? I think it is but I want to hear more opinions.

**Comment by mxaddict** [comment]
I'd say it's a move in the right direction, the UI needs to be updated to look better though, I've done a little of that, but can definitely be improved.

**Comment by mxaddict** [comment]
Maybe we should ask on discord about it

File: src/qt/navcoingui.cpp [hunk 1]
```diff
@@ -416,6 +418,20 @@ void NavCoinGUI::createActions()
     historyAction->setShortcut(QKeySequence(Qt::ALT + Qt::Key_4));
     tabGroup->addAction(historyAction);
 
+    daoAction = new QAction(platformStyle->Icon(":/icons/dao"), tr("&Transactions"), this);
+    daoAction->setStatusTip(tr("Participate in the DAO"));
+    daoAction->setToolTip(daoAction->statusTip());
+    daoAction->setCheckable(true);
+    daoAction->setShortcut(QKeySequence(Qt::ALT + Qt::Key_4));
```
[the remark this answers is not in the document]
Comment (aguycalled) [reply src/qt/navcoingui.cpp:431]: this shortcut is still duplicated

**Comment by navbuilder** [comment]
A new build of a7418ad47d97f62f7ec529036c11ff3b1287898d has completed succesfully!
Binaries available at https://build.nav.community/binaries/moved-options-dialog

**Comment by mxaddict** [comment]
@chasingkirkjufell @aguycalled I've fixed the shortcut key issue, this should be ready for final review.

**Comment by chasingkirkjufell** [comment]
Any changes to the options wouldn't stick. Like if I enable the coin control, it wouldn't show that. And going to another tab then back to setting tab then I would see the option back to previous setting.

Looks really good though. I quite like it. :) Needed some time to be sure that I like it.

**Comment by mxaddict** [comment]
> Any changes to the options wouldn't stick. Like if I enable the coin control, it wouldn't show that. And going to another tab then back to setting tab then I would see the option back to previous setting.
> 
> Looks really good though. I quite like it. :) Needed some time to be sure that I like it.

I'm not sure what you mean, the coin control setting will be overriden if you also change it on the send page (On send page it's autosave, so it will be on/off depending on if you ticked it on/off on the send page)

As for the options page, you have to explicitly click on save, otherwise changes are discarded if you go to another tab.

**Comment by mxaddict** [comment]
@chasingkirkjufell I've added the confirmation dialog for when the user has not saved the settings and moves to a different page

**Comment by navbuilder** [comment]
A new build of a6b02d262a2f085102e1b0b873f20a4e81573272 has completed succesfully!
Binaries available at https://build.nav.community/binaries/moved-options-dialog

**Comment by chasingkirkjufell** [comment]
For the notifications, may i suggest
"You have not saved your changes, are you sure you want to discard them?"
and 
"Changes have been saved."?

**Comment by mxaddict** [comment]
@chasingkirkjufell I've made the 2 text changes suggested 👍

**Comment by navbuilder** [comment]
A new build of 01c424a043495b0761d85f38548a7b11fade4a8e has completed succesfully!
Binaries available at https://build.nav.community/binaries/moved-options-dialog

**Pull request merged**